org.hl7.fhir.convertors/src/main/java/org/hl7/fhir/convertors/VersionConvertorResourceNameMapper.java

```diff
@@ -0,0 +1,767 @@
+    if (name.equals("Sequence")) { return null; }
+    if (name.equals("NamingSystem")) { return "NamingSystem"; }
+    if (name.equals("NutritionOrder")) { return "NutritionOrder"; }
+    if (name.equals("Observation")) { return "Observation"; }
+    if (name.equals("OperationDefinition")) { return "OperationDefinition"; }
+    if (name.equals("OperationOutcome")) { return "OperationOutcome"; }
+    if (name.equals("Task")) { return "Order"; }
+    // if (name.equals("Task")) { return "OrderResponse"; }
+    if (name.equals("Organization")) { return "Organization"; }
+    if (name.equals("Parameters")) { return "Parameters"; }
+    if (name.equals("Patient")) { return "Patient"; }
+    if (name.equals("PaymentNotice")) { return "PaymentNotice"; }
+    if (name.equals("PaymentReconciliation")) { return "PaymentReconciliation"; }
+    if (name.equals("Person")) { return "Person"; }
+    if (name.equals("PlanDefinition")) { return null; }
+    if (name.equals("Practitioner")) { return "Practitioner"; }
+    if (name.equals("PractitionerRole")) { return null; }
+    if (name.equals("Procedure")) { return "Procedure"; }
+    if (name.equals("ProcessRequest")) { return "ProcessRequest"; }
+    if (name.equals("ProcessResponse")) { return "ProcessResponse"; }
+    if (name.equals("Provenance")) { return "Provenance"; }
+    if (name.equals("Questionnaire")) { return "Questionnaire"; }
+    if (name.equals("QuestionnaireResponse")) { return "QuestionnaireResponse"; }
+    if (name.equals("ReferralRequest")) { return "ReferralRequest"; }
+    if (name.equals("RelatedPerson")) { return "RelatedPerson"; }
+    if (name.equals("RequestGroup")) { return null; }
+    if (name.equals("ResearchStudy")) { return null; }
+    if (name.equals("ResearchSubject")) { return null; }
+    if (name.equals("RiskAssessment")) { return "RiskAssessment"; }
+    if (name.equals("Schedule")) { return "Schedule"; }
+    if (name.equals("SearchParameter")) { return "SearchParameter"; }
+    if (name.equals("ServiceDefinition")) { return null; }
+    if (name.equals("ProcedureRequest")) { return "ProcedureRequest"; }
+    if (name.equals("Slot")) { return "Slot"; }
+    if (name.equals("Specimen")) { return "Specimen"; }
+    if (name.equals("StructureDefinition")) { return "StructureDefinition"; }
+    if (name.equals("StructureMap")) { return null; }
+    if (name.equals("Subscription")) { return "Subscription"; }
+    if (name.equals("Substance")) { return "Substance"; }
+    if (name.equals("SupplyDelivery")) { return "SupplyDelivery"; }
+    if (name.equals("SupplyRequest")) { return "SupplyRequest"; }
+    if (name.equals("Task")) { return null; }
+    if (name.equals("TestReport")) { return null; }
+    if (name.equals("TestScript")) { return "TestScript"; }
+    if (name.equals("ValueSet")) { return "ValueSet"; }
+    if (name.equals("VisionPrescription")) { return "VisionPrescription"; }
+    return null;
+  }
+
+
+}

```

Comment:
So, for something like this with so many if conditions, I would lean towards using a switch statement instead:

    

      switch (name) {

      case "AllergyIntolerance": return "AllergyIntolerance";

      case "Appointment": return "Appointment";

        ...
      default: return null;

    }

From a readability perspective, it's a lot less code and easier to parse visually, and switch statements are much more performant than if conditions, especially when the number of choices is large.

One other thing that you could do that would really clean it up and remove all the "..." Strings would be to use <class>.class.getName() for the types you were checking and returning. ex:

`case org.hl7.fhir.dstu2.Appointment.class.getName(): return org.hl7.fhir.dstu3.Appointment.class.getName();`

but I understand if that is too tedious. It just takes out any magic Strings, which aren't great.

Reply:
you can do switch statements on strings? since when? 

But these absolutely are magic strings...

Reply:
Did some of the changes. but the case statements have to be literal strings